### PR TITLE
Feature/api server tokens

### PR DESCRIPTION
**Pull request by OBorce**
- new endpoints for fetching tokens and nft data
- keep track of token balances in the DB

### CODE REVIEW

File: api-server/api-server-common/src/storage/storage_api/mod.rs [hunk 1]
```diff
@@ -136,6 +140,56 @@ impl Utxo {
...
+    }
+
+    pub fn change_authority(mut self, authority: Destination) -> Self {
+        self.authority = authority;
+        self
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode)]
+pub enum CoinOrTokenId {
```
Comment (azarovh) [inline api-server/api-server-common/src/storage/storage_api/mod.rs:188]: I recently moved this struct to `common/primitives` so you can reuse it now

File: api-server/scanner-lib/src/blockchain_state/mod.rs [hunk 2]
```diff
@@ -544,15 +583,60 @@ async fn update_tables_from_transaction_inputs<T: ApiServerStorageWrite>(
...
     let mut address_transactions: BTreeMap<Address<Destination>, BTreeSet<Id<Transaction>>> =
         BTreeMap::new();
 
     for input in inputs {
         match input {
-            // TODO: update token states
-            | TxInput::AccountCommand(_, _) => {}
+            TxInput::AccountCommand(_, cmd) => match cmd {
+                AccountCommand::MintTokens(token_id, amount) => {
+                    let issance =
```
Comment (azarovh) [inline api-server/scanner-lib/src/blockchain_state/mod.rs:595]: typo `issance` several times

**Review by azarovh — APPROVED** [review]

File: api-server/api-server-common/src/storage/storage_api/mod.rs (hunk 1 above)
Comment (OBorce) [reply api-server/api-server-common/src/storage/storage_api/mod.rs:188]: I saw that, but was not sure if I should add encode/decode to it

**Review by TheQuantumPhysicist — APPROVED** [review]

File: api-server/scanner-lib/src/blockchain_state/mod.rs (hunk 2 above)
Comment (OBorce) [reply api-server/scanner-lib/src/blockchain_state/mod.rs]: fixed

File: api-server/api-server-common/src/storage/storage_api/mod.rs (hunk 1 above)
Comment (OBorce) [reply api-server/api-server-common/src/storage/storage_api/mod.rs]: added encode/decode and reused the existing enum

**Pull request merged**